### PR TITLE
Update k8s.io deps to 1.31.0

### PR DESCRIPTION
**Pull request by tigrato**
Depends on #46299

### CODE REVIEW

File: integrations/operator/crdgen/testdata/protofiles/teleport/legacy/client/proto/event.proto [hunk 1]
```diff
@@ -175,7 +178,11 @@ message Event {
     teleport.clusterconfig.v1.AccessGraphSettings AccessGraphSettings = 61;
     // SPIFFEFederation is a resource for SPIFFE federation.
     teleport.machineid.v1.SPIFFEFederation SPIFFEFederation = 62;
-    // StaticHostUser is a resource for static host users.
-    teleport.userprovisioning.v1.StaticHostUser StaticHostUser = 63;
+    // AutoUpdateConfig is a resource for autoupdate config.
```
Comment (codingllama) [inline integrations/operator/crdgen/testdata/protofiles/teleport/legacy/client/proto/event.proto:181]: Why are the proto changes necessary? Are they part of the fix for k8s.io v1.31.x?

File: examples/chart/teleport-cluster/charts/teleport-operator/operator-crds/resources.teleport.dev_accesslists.yaml [hunk 2]
```diff
@@ -194,16 +194,8 @@ spec:
...
-                    state of this API Resource.\n---\nThis struct is intended for
-                    direct use as an array at the field path .status.conditions.  For
-                    example,\n\n\n\ttype FooStatus struct{\n\t    // Represents the
-                    observations of a foo's current state.\n\t    // Known .status.conditions.type
-                    are: \"Available\", \"Progressing\", and \"Degraded\"\n\t    //
-                    +patchMergeKey=type\n\t    // +patchStrategy=merge\n\t    // +listType=map\n\t
-                    \   // +listMapKey=type\n\t    Conditions []metav1.Condition `json:\"conditions,omitempty\"
-                    patchStrategy:\"merge\" patchMergeKey:\"type\" protobuf:\"bytes,1,rep,name=conditions\"`\n\n\n\t
-                    \   // other fields\n\t}"
+                  description: Condition contains details for one aspect of the current
```
Comment (codingllama) [inline examples/chart/teleport-cluster/charts/teleport-operator/operator-crds/resources.teleport.dev_accesslists.yaml:197]: What causes all of these yaml changes?

**Review by codingllama — COMMENTED** [review]
Thanks, Tiago!

File: examples/chart/teleport-cluster/charts/teleport-operator/operator-crds/resources.teleport.dev_accesslists.yaml (hunk 2 above)
Comment (hugoShaka) [reply examples/chart/teleport-cluster/charts/teleport-operator/operator-crds/resources.teleport.dev_accesslists.yaml:197]: The CRDs are generated from two sources:
- CR spec comes from our proto definitions
- CR status comes from go structs importing kubernetes structs like the conditions. The descriptions are coming from the godoc of those upstream structs

When we update the kube version, we also update those structs. If they changed the godocs, it triggers those changes.

tl;dr it's safe an expected, I went through the diffs and it's only on status fields coming from upstream

**Review by hugoShaka — APPROVED** [review]

File: integrations/operator/crdgen/testdata/protofiles/teleport/legacy/client/proto/event.proto (hunk 1 above)
Comment (hugoShaka) [reply integrations/operator/crdgen/testdata/protofiles/teleport/legacy/client/proto/event.proto:181]: Tiago re-syced the test snapshots fopr the CRD generation. We use a copy of the proto definitions rather than the real ones in the snapshot tests. It's dumb and not useful but I never got time to clean this up, so we need to sync the test inputs from time to time.

Comment (codingllama) [reply integrations/operator/crdgen/testdata/protofiles/teleport/legacy/client/proto/event.proto:181]: This is fine, but I want to understand what is part of the k8s v1.31.x fix and what is just tagging along.

File: examples/chart/teleport-cluster/charts/teleport-operator/operator-crds/resources.teleport.dev_accesslists.yaml (hunk 2 above)
Comment (codingllama) [reply examples/chart/teleport-cluster/charts/teleport-operator/operator-crds/resources.teleport.dev_accesslists.yaml:197]: So this is a consequence of the k8s version bump? Got it.

Comment (tigrato) [reply examples/chart/teleport-cluster/charts/teleport-operator/operator-crds/resources.teleport.dev_accesslists.yaml:197]: Yes. `sigs.k8s.io/controller-tools` is responsible for generating the CRD descriptions and auto-added these fixed descriptions to the `status` field of the objects.

In the most recent version, the change removed the extra and unrelated description. Seems much better now that it doesn't reference FooStatus 🙃

File: integrations/operator/crdgen/testdata/protofiles/teleport/legacy/client/proto/event.proto (hunk 1 above)
Comment (tigrato) [reply integrations/operator/crdgen/testdata/protofiles/teleport/legacy/client/proto/event.proto:181]: The protobuf updates are unrelated but since I generated the new CRD stubs, I also updated the protos.

**Review by codingllama — APPROVED** [review]

**Comment by codingllama** [comment]
(Sorry for the delay, I thought had approved this already.)
